CardForm/src/main/java/com/braintreepayments/cardform/utils/CardNumberTransformation.java

```diff
@@ -0,0 +1,38 @@
+package com.braintreepayments.cardform.utils;
+
+import android.graphics.Rect;
+import android.text.method.TransformationMethod;
+import android.view.View;
+
+import java.util.Arrays;
+
+/**
+ * A transformation that masks card numbers up to the last 4 digits. For example, it will transform
+ * "4111111111111111" to "●●●● 1111". This can be used to mask card numbers in an {@link android.widget.EditText}.
+ */
+public class CardNumberTransformation implements TransformationMethod {
+
+    private static final String FOUR_DOTS = "••••";
+
+    @Override
+    public CharSequence getTransformation(final CharSequence source, View view) {
+        if (source.length() >= 9) {
+            StringBuilder result = new StringBuilder()
+                    .append(FOUR_DOTS)
+                    .append(" ")
+                    .append(source.subSequence(source.length() - 4, source.length()));
+
+            char[] padding = new char[source.length() - result.length()];
+            Arrays.fill(padding, Character.MIN_VALUE);
```

Comment:
It might be clearer/look better if we actually masked every character instead of just dropping 4 ●s and the last four. It seems like some of your number is disappearing then.

Reply:
Hmm, to my eye this is _slightly_ cleaner after trying it both ways, and it mimics what iOS does.